### PR TITLE
Refactor `as_rtf()`

### PR DESCRIPTION
**Pull request by yihui**
Massive refactoring of `as_rtf()` to share code with `as_gt()`.

### CODE REVIEW

File: R/as_rtf.R
Comment (yihui) [reply R/as_rtf.R]: Now I think we should do it the other way around, i.e., we should make `as_gt()` consistent with `as_rtf()` by hardcoding the string since it seems `"Null hypothesis"` is hardcoded in `x`. Correct me if I'm wrong.

Comment (yihui) [reply R/as_rtf.R]: Forget it. I think I was wrong. Changing it will break a test.

File: R/as_rtf.R
Comment (yihui) [reply R/as_rtf.R]: I added this line for `as_rtf()`, otherwise this argument would be meaningless. This also makes it consistent with `as_gt()`.

File: R/as_gt.R
Comment (jdblischak) [inline R/as_gt.R:377]: Might be worth documenting the argument `transform` since it isn't as straight-forward as the others

**Review by jdblischak — APPROVED** [review]

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:377]: Absolutely. I should document these new functions. Thanks for the suggestion!

File: R/as_gt.R
Comment (yihui) [inline R/as_gt.R:246]: @LittleBeannie I have a question regarding a difference between `as_gt()` and `as_rtf()` that I noticed while comparing the two: for `as_gt()`, we obtain `full_alpha` from the attribute of `x`, whereas for `as_rtf()`, `full_alpha` is its argument and not retrieved from `x`'s attributes. Is this expected? I'm not sure if you or @elong0527 is the right person to ask.

File: R/as_gt.R
Comment (LittleBeannie) [inline R/as_gt.R:93]: Is it appropriate to refer to this as the 'fixed_design_method'? From my perspective, the term 'fixed method' implies that the method is fixed, which may have limited clinical significance.

Furthermore, I've observed utility functions for both fixed design (such as `fixed_method`, `fixed_title`, `method_footnote`), as well as group sequential design (such as `get_display_columns `, `footnote_content`, `footnote_non_binding`, `footnote_row`, `gs_design_info`). I am considering whether it would be beneficial to use distinct prefixes for these two sets of utilities to differentiate their purposes.

Finally, I am wondering if we should save the above 2 sets of utilities into a separate R file, and name it as `utility_as_gt`.

File: R/as_gt.R
Comment (LittleBeannie) [inline R/as_gt.R:99]: Echo with my previous comment, can we call this function as `fixed_design_title`?

File: R/as_gt.R
Comment (LittleBeannie) [inline R/as_gt.R:110]: I am wondering why the prefix of `fixed_` is missing... Is the `method_footnote` appliable for both fixed design and group sequential design?

File: R/as_gt.R
Comment (LittleBeannie) [reply R/as_gt.R:246]: Hi @yihui , yes, I can answer the question. The `full_alpha` should be retrieved from `x`'s attributes. I will prepare a PR to get it there.

File: R/as_gt.R
Comment (LittleBeannie) [inline R/as_gt.R:379]: I am hesitant to use `gs_design_info` as the name, as `info` in gsDesign2 carries a specific meaning – statistical information (1/variance). Can we consider an alternative name for it?

**Review by LittleBeannie — COMMENTED** [review]
Hi @yihui, there are lots of work, thank you so much! I only have some very minor comments, regarding the naming of the utility functions.

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:246]: If you want to do it, please wait until this PR is merged, or I can just do it in this PR.

Comment (LittleBeannie) [reply R/as_gt.R:246]: Thanks for the confirmation. I know there are many changes occurring in this pull request, so let's wait until this PR is merged.

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:93]: > I am considering whether it would be beneficial to use distinct prefixes for these two sets of utilities to differentiate their purposes.

I'm using the prefixes `fd_` and `gsd_` now, and am open to other suggestions. I tend not to use the prefix `gs_design_` because I noticed that we have had functions with this prefix but serve a very different purpose. I hope the abbreviations `fd` and `gsd` make sense. They are only for ourselves and users won't use them, so the naming is perhaps not critically important. Anyway, naming is really hard :)

> Finally, I am wondering if we should save the above 2 sets of utilities into a separate R file, and name it as `utility_as_gt`.

I don't have an opinion on that. I can certainly put these functions in `as_utils.R` if you prefer.

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:110]: I was simply being lazy. It was only for fixed design.

File: R/as_gt.R
Comment (yihui) [reply R/as_gt.R:379]: The naming of this function is the hardest. I spent nearly half an hour on it but couldn't think of a better name. I ended up using `gsd_parts()`. Please feel free to suggestion a different one.

Comment (LittleBeannie) [reply R/as_gt.R]: Agree, naming is always the hardest part. As long as we could get rid of the `info` as keywords, we should be good. The new name `gsd_parts` looks good to me.

File: R/as_gt.R
Comment (LittleBeannie) [reply R/as_gt.R]: I like the `fd` and `gsd`, which are sufficiently clear for developers. I am also good for the current folder/file structure. I will get it merged. Thanks for all the work!

**Review by LittleBeannie — APPROVED** [review]

**Pull request merged**
